src/plot_api/plot_api.js

```diff
@@ -2887,23 +2887,28 @@ function lsInner(gd) {
                 .call(Color.fill, fullLayout.plot_bgcolor);
         }
 
+
         // Clip so that data only shows up on the plot area.
         var clips = fullLayout._defs.selectAll('g.clips'),
             clipId = 'clip' + fullLayout._uid + subplot + 'plot';
 
-        clips.selectAll('#' + clipId)
-            .data([0])
-        .enter().append('clipPath')
+        var plotClip = clips.selectAll('#' + clipId)
+            .data([0]);
+
+        plotClip.enter().append('clipPath')
             .attr({
                 'class': 'plotclip',
                 'id': clipId
             })
-            .append('rect')
+            .append('rect');
+
+        plotClip.selectAll('rect')
```

Comment:
Looks like d3 v4 will make this easier.

Taken from https://medium.com/@mbostock/what-makes-software-good-943557f8a488#.rb39u8abk

![image](https://cloud.githubusercontent.com/assets/6675409/14893413/5957c2f0-0d3d-11e6-9338-9e2c4feb6610.png)